kingpin/actors/rightscale/deployment.py

```diff
@@ -98,13 +112,43 @@ def _execute(self):
             self.log.info('Would create a deployment %s' % self.option('name'))
             self.log.debug('Deployment params: %s' % params)
             raise gen.Return()
-        self.log.info('Deployment params: %s' % params)
-
-        try:
-            yield self._client.create_resource(
-                self._client._client.deployments, params)
-        except Exception as e:
-            self.log.critical('Encountered error: %s' % e)
-            raise exceptions.UnrecoverableActorFailure(
-                'Could not create deployment %s. Does it exist already?' %
-                self.option('name'))
+
+        self.log.info('Creating deployment %s' % self.option('name'))
+
+        yield self._client.create_resource(
+            self._client._client.deployments, params)
+
+
+class Destroy(DeploymentBaseActor):
+
+    """Deletes a RightScale deployment.
+
+    Options match the documentation in RightScale:
+    http://reference.rightscale.com/api1.5/resources/ResourceDeployments.html
+
+    **Options**
+
+    :name:
+      The name of the deployment to be deleted.
+    """
+
+    all_options = {
+        'name': (str, REQUIRED, 'The name of the deployment to be deleted.'),
+    }
+
+    @gen.coroutine
+    def _execute(self):
+
+        dep = yield self._find_deployment(self.option('name'))
+        if not dep:
+            raise exceptions.InvalidOptions(
+                'Deployment "%s" does not exist.' % self.option('name'))
+
+        info = (yield self._client.show(dep.self)).soul
+
+        if self._dry:
+            self.log.info('Would delete deployment %s' % info['name'])
+            raise gen.Return()
+
+        self.log.info('Deleting deployment %s' % info['name'])
+        yield self._client.destroy_resource(dep)
```

Comment:
Something just occurred to me as I was reading this -- what happens if we try to delete a deployment with servers in it? Is an error returned back to us? Can you try that out real quick?